### PR TITLE
feat: introduce is_public in session-speaker wizard

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5100
server : https://github.com/fossasia/open-event-server/pull/7524

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Before this, we need to create a migration to make previous fields as public. And decide which ones should be public by default

**Comment by maze-runnar** [comment]
> we need to create a migration to make previous fields as public.

is it would be done like - like op.execute("update custom_forms set is_public = true where condition)

**Comment by iamareebjamal** [comment]
Yes

**Comment by maze-runnar** [comment]
how to add a migration? Means I have already added isPublic in schema previously so how to migrate now, to make previous fields as public.

**Comment by iamareebjamal** [comment]
You wrote the answer yourself in the comment above

`op.execute("update custom_forms set is_public = true where condition")`

Find the field identifiers which should be public and then add them in the condition clause

**Comment by maze-runnar** [comment]
I am asking for where to write op.execute("update custom_forms set is_public = true where condition"). Because I am not making any changes in schema at the time. So no migration file will be generated. 😕

**Comment by iamareebjamal** [comment]
Whenever you need help with the command, you run it without arguments or with `--help`

```
python manage.py db
```

```
usage: Perform database migrations

Perform database migrations

positional arguments:
  {init,revision,migrate,edit,merge,upgrade,downgrade,show,history,heads,branches,current,stamp}
    init                Creates a new migration repository
    revision            Create a new revision file.
    migrate             Alias for 'revision --autogenerate'
    edit                Edit current revision.
    merge               Merge two revisions together. Creates a new migration file
    upgrade             Upgrade to a later version
    downgrade           Revert to a previous version
    show                Show the revision denoted by the given symbol.
    history             List changeset scripts in chronological order.
    heads               Show current available heads in the script directory
    branches            Show current branch points
    current             Display the current revision for each database.
    stamp               'stamp' the revision table with the given revision; don't run any migrations

optional arguments:
  -?, --help            show this help message and exit
```

Which shows it can be done using `revision` command

File: app/mixins/custom-form.js [hunk 1]
```diff
@@ -19,6 +20,7 @@ export default Mixin.create(MutableArray, {
         form            : 'session',
         type            : 'text',
         isRequired      : false,
+        isPublic        : false,
```
Comment (iamareebjamal) [inline app/mixins/custom-form.js:23]: Should be public

File: app/mixins/custom-form.js [hunk 2]
```diff
@@ -35,6 +38,7 @@ export default Mixin.create(MutableArray, {
...
```
Comment (iamareebjamal) [inline app/mixins/custom-form.js:41]: Should be public

**Comment by iamareebjamal** [comment]
@maze-runnar Please complete this PR as instructed in the server PR. All the fields I have marked public in that PR should be public here

**Comment by maze-runnar** [comment]
ok

File: app/mixins/custom-form.js [hunk 3]
```diff
@@ -43,6 +47,7 @@ export default Mixin.create(MutableArray, {
...
```
Comment (iamareebjamal) [inline app/mixins/custom-form.js:50]: This is also public IIRC

File: app/mixins/custom-form.js [hunk 4]
```diff
@@ -134,6 +149,7 @@ export default Mixin.create(MutableArray, {
         form            : 'speaker',
         type            : 'text',
         isRequired      : false,
+        isPublic        : false,
```
Comment (iamareebjamal) [inline app/mixins/custom-form.js:152]: This was public

**Comment by mariobehling** [comment]
This is only implementing the public vs private information. However we still do not have a private or public page for the session or speaker info where we can clearly see the information. I am trying to make the issue clearer. As we come back to this over and over again, please help to resolve this until the end. So, we do not loose the train of thought and can resolve this ultimately: https://github.com/fossasia/open-event-frontend/issues/6222

**Pull request merged**